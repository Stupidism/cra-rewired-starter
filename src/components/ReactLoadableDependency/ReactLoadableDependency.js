import React from 'react';
import Loading from 'modules/Loading/index';
import ExternalLink from 'modules/ExternalLink/index';

const links = {
  reactLoadable: (
    <ExternalLink to="https://github.com/jamiebuilds/react-loadable/">
      react-loadable
    </ExternalLink>
  ),
};

const ReactLoadableDependency = () => (
  <div>
    <p>
      Using {links.reactLoadable}, code-splits for Routes and Components are
      easy peasy.
    </p>
    <Loading />
    <p>You can see this loading, because it's intentionally delayed.</p>
  </div>
);

export default ReactLoadableDependency;
