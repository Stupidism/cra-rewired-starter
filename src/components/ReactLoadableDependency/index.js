import ReactLoadableDependency from './ReactLoadableDependency';

export default ReactLoadableDependency;
