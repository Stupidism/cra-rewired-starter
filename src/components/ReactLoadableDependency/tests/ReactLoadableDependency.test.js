import React from 'react';
import { shallow } from 'enzyme';

import ReactLoadableDependency from '../ReactLoadableDependency';

describe('<ReactLoadableDependency />', () => {
  it('should be consistent with snapshot', () => {
    const snapshot = shallow(<ReactLoadableDependency />);
    expect(snapshot).toMatchSnapshot();
  });
});
