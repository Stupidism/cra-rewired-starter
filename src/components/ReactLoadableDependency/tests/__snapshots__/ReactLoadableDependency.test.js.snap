// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactLoadableDependency /> should be consistent with snapshot 1`] = `
<div>
  <p>
    Using 
    <ExternalLink
      to="https://github.com/jamiebuilds/react-loadable/"
    >
      react-loadable
    </ExternalLink>
    , code-splits for Routes and Components are easy peasy.
  </p>
  <Loading />
  <p>
    You can see this loading, because it's intentionally delayed.
  </p>
</div>
`;
